src/Storages/MergeTree/MergeTreeIndexConditionText.cpp

```diff
@@ -483,18 +488,32 @@ bool MergeTreeIndexConditionText::traverseFunctionNode(
         if (!value_data_type.isStringOrFixedString())
             return false;
 
-        /// mapContainsKey can be used only with an index defined as `mapKeys(Map(String, ...))`
-        /// mapContainsValue can be used only with an index defined as `mapValues(Map(String, ...))`
-        bool is_supported_key_function = has_map_keys_column && (function_name == "mapContainsKey" || function_name == "has");
-        bool is_supported_value_function = has_map_values_column && function_name == "mapContainsValue";
-
-        if (is_supported_key_function || is_supported_value_function)
+        auto use = [&](auto tokens) -> bool
```

Comment:
We can pass the function here to retain the behavior with other functions. Also, we can rename to be more explicit.
```suggestion
        auto make_map_function = [&](RPNElement::Function function, auto tokens) -> bool
```